library/src/main/java/com/getstream/sdk/chat/utils/MessageInputController.kt

```diff
@@ -338,90 +332,30 @@ class MessageInputController(private val context: Context,
-		val title = binding.tvTitle.context.resources.getString(if (isCommand) R.string.stream_input_type_command else R.string.stream_input_type_auto_mention)
-		binding.tvTitle.text = title
-		binding.tvCommand.text = ""
-		setCommandMentionListItemAdapter(isCommand)
-		openCommandView()
-		binding.lvCommand.onItemClickListener = AdapterView.OnItemClickListener { adapterView: AdapterView<*>?, view: View?, position: Int, l: Long ->
-			if (isCommand) binding.etMessage.setText("/" + (commands !![position] as Command).name + " ") else {
-				val messageStr = binding.etMessage.text.toString()
-				val userName = (commands !![position] as User).getExtraValue("name", "")
-				val converted = StringUtility.convertMentionedText(messageStr, userName)
-				binding.etMessage.setText(converted)
-			}
-			binding.etMessage.setSelection(binding.etMessage.text.length)
-			closeCommandView()
+	fun checkCommandsOrMentions(inputMessage: String) {
+		when {
+			inputMessage.isCommandMessage() -> { view.showSuggestedCommand(channelCommands.matchName(inputMessage.removePrefix("/"))) }
+			inputMessage.isMentionMessage() -> { view.showSuggestedMentions(members.matchUserName(inputMessage.substringAfterLast("@"))) }
+			else -> { cleanSuggestion() }
 		}
 	}
 
-	private fun setCommandMentionListItemAdapter(isCommand: Boolean) {
-		if (commandMentionListItemAdapter == null) {
-			commandMentionListItemAdapter = CommandMentionListItemAdapter<MessageInputStyle>(context, commands, style, isCommand)
-			binding.lvCommand.adapter = commandMentionListItemAdapter
-		} else {
-			commandMentionListItemAdapter !!.setCommand(isCommand)
-			commandMentionListItemAdapter !!.setCommands(commands)
-			commandMentionListItemAdapter !!.notifyDataSetChanged()
-		}
+	private fun cleanSuggestion() {
+		view.showSuggestedMentions(listOf())
+		view.showSuggestedCommand(listOf())
 	}
 
-	private fun setCommandsMentionUsers(string: String) {
-		if (commands == null) commands = ArrayList()
-		commands !!.clear()
-		if (string.startsWith("/")) {
-			if (channelCommands.isEmpty()) return
-			val commandStr = string.replace("/", "")
-			setCommands(commandStr)
-			binding.tvCommand.text = commandStr
-		} else {
-			val names = string.split("@").toTypedArray()
-			if (names.isNotEmpty()) setMentionUsers(names[names.size - 1])
-		}
+	fun onCommandSelected(command: Command) {
+		view.messageText = "/${command.name} "
 	}
 
-	private fun setCommands(commandPattern: String) {
-		commands = channelCommands.filter { it.name.contains(commandPattern) }.toMutableList()
+	fun onUserSelected(currentMessage: String, user: User) {
+		view.messageText = "${currentMessage.substringBeforeLast("@")}@${user.name} "
 	}
+}
 
-	private fun setMentionUsers(namePattern: String) {
-		commands = members.map { it.user }
-				.filter { it.getExtraValue("name", "").contains(namePattern, true) }
-				.toMutableList()
-	}
-}
\ No newline at end of file
+private fun String.isCommandMessage() = COMMAND_PATTERN.matcher(this).find()
```

Comment:
`private vals` on the top, `private funs` on the bottom: looks not aligned. Do you follow some style?

Reply:
In that case I am putting "Constant values" on top of the class and "Private fun extensions" on the bottom, It has been useful for me in previous jobs, but we can change it 
I'm not sure if you have any defined style, if not, maybe we can work on a common style and would be great to use some checkstyle tool like [ktlint](https://ktlint.github.io/)